### ISSUE TITLE
Does this method work with the pages router?

### ISSUE THREAD

**Issue by remusris** [digitros/nextjs-fastapi#2]
Can you copy the code from next.config.ts in an existing pages router nextJS project to get this method to work?

**Comment by digitros** [comment]
For sure. Just make sure you change the path to the python file in the package.json script.
Something like this, if you're not using "src": 
`"fastapi-dev": "pip3 install -r requirements.txt && python3 -m uvicorn pages.api.index:app --reload",`

**Comment by remusris** [comment]
My second question is can this be used alongside trpc? I'm referring to a t3 stack scaffold

**Comment by IdoPesok** [comment]
@digitros will it work with tRPC ? ^

**Comment by Wolfleader101** [comment]
Just following up on if anyone got this working

**Comment by Wolfleader101** [comment]
> @digitros will it work with tRPC ? ^
> 
> I got it to work correctly on my localhost but for some reason Vercel isn't recognizing the Python file in src/pages/api/[dir name]

I've been update to replicate on localhost, it does not rewrite for me

**Comment by JohnVersus** [comment]
It should work with the app folder too, but you can't add the file inside app folder. Create a new`api` folder in the root and add your python function API folder along with index.py. 

reference: https://vercel.com/docs/functions/serverless-functions/runtimes
<img width="433" alt="image" src="https://github.com/digitros/nextjs-fastapi/assets/15834299/7f89350c-a6e8-4715-a302-b6fb6282620a">

**Comment by Wolfleader101** [comment]
> It should work with the app folder too, but you can't add the file inside app folder. Create a new`api` folder in the root and add your python function API folder along with index.py.
> 
> reference: https://vercel.com/docs/functions/serverless-functions/runtimes <img alt="image" width="433" src="https://user-images.githubusercontent.com/15834299/268344293-7f89350c-a6e8-4715-a302-b6fb6282620a.png">

I haven't gotten it to work with the pages directory.

I have
![image](https://github.com/digitros/nextjs-fastapi/assets/25214835/526dbc60-acc9-4da4-a518-3e812b959a71)
![image](https://github.com/digitros/nextjs-fastapi/assets/25214835/ec2db0fd-211e-4969-b37e-aadc84c522d3)

```js
  rewrites: async () => {
    return [
      {
        source: "/api/py/:path*",
        destination:
          process.env.NODE_ENV === "development"
            ? "http://127.0.0.1:8000/api/py/:path*"
            : "/api/py",
      },
    ];
  },```

it doesn't seem to be deploying a python API on vercel

**Comment by JohnVersus** [comment]
@Wolfleader101  It works the same way in pages too. 

Since you mentioned `/api/py` route as prefix in code, vercel will not use it for the path. Vercel only uses the file name or folder name for the API path

As per your current setup your python code should run in `api/index/api/py`

**Comment by Wolfleader101** [comment]
> api

It seems to be that it works but only on a per file basis - you can't have a fully setup fast API router, you need to have one per file...
Any solution to get past this?

**Comment by JohnVersus** [comment]
> > api
> 
> It seems to be that it works but only on a per file basis - you can't have a fully setup fast API router, you need to have one per file... Any solution to get past this?

Yeah, I think that's a limitation when you are using Python runtime with the next js framework. 

To fully utilize the Python runtime like one single Python backend app you need to remove the next js files and host it with just vercel config.

**Comment by Wolfleader101** [comment]
> > > api
> > 
> > 
> > It seems to be that it works but only on a per file basis - you can't have a fully setup fast API router, you need to have one per file... Any solution to get past this?
> 
> Yeah, I think that's a limitation when you are using Python runtime with the next js framework.
> 
> To fully utilize the Python runtime like one single Python backend app you need to remove the next js files and host it with just vercel config.

Yea seems to be a limitation with multiple project types. Hopefully they fix it in the future.
